src/tfr_executive/src/teleop_action_server.cpp

```diff
@@ -97,58 +97,115 @@ class TeleopExecutive
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::STOP_TURNTABLE):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, STOP_TURNTABLE");
+                        //all zeros by default
+                        //TODO integrate manual turntable control
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::FORWARD):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, FORWARD");
+                        move_cmd.linear.x = drive_stats.getLinear();
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::BACKWARD):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, BACKWARD");
+                        move_cmd.linear.x = -drive_stats.getLinear();
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::LEFT):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, LEFT");
+                        move_cmd.angular.z = drive_stats.getAngular();
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::RIGHT):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, RIGHT");
+                        move_cmd.angular.z = -drive_stats.getAngular();
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::CLOCKWISE):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, CLOCKWISE");
+                        move_cmd.angular.z = drive_stats.getAngular();
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::COUNTERCLOCKWISE):
+                    {
+                        ROS_INFO("Teleop Action Server: Command Recieved, COUNTERCLOCKWISE");
+                        move_cmd.angular.z = -drive_stats.getAngular();
+                        drivebase.publish(move_cmd);
+                        break;
+                    }
+
+                case (tfr_utilities::TeleopCode::DIG):
+                    {
+                        // TODO integrate and check for preemption
+                        ROS_INFO("Teleop Action Server: Command Recieved, DIG");
+                        tfr_msgs::DurationSrv digging_time;
+                        ros::service::call("digging_time", digging_time);
```

Comment:
I give it the time allowed to dig just like the specification

Reply:
@AdamuKaapan Executive should just know how to tell it to go with a timer. The Mining System is in charge of deciding how many scoops that is. This looks correct to me.

Reply:
Yeah, that makes sense; I just wasn't sure if we wanted to do it a bit more granularly because it's manual operation, but I'm totally fine with this too.